### PR TITLE
Create personal team on account creation

### PR DESCRIPTION
**Pull request by cjdenio**
This PR does that and also removes the `POST /api/users` endpoint

### CODE REVIEW

File: pkg/api/oauth/oauth.go [hunk 1]
```diff
@@ -86,6 +86,14 @@ func SetupRoutes(r *gin.RouterGroup) {
 				SlackUserID: resp.AuthedUser.ID,
 				Name:        identity.User.Name,
 				Avatar:      identity.User.Image512,
+				Teams: []*db.Team{
+					{
+						Name:      "Personal team",
+						Automatic: false,
+						Personal:  true,
+						HNUserID:  "haas_" + resp.AuthedUser.ID,
```
Comment (anirudhb) [inline pkg/api/oauth/oauth.go]: Add a FIXME here

File: pkg/db/users.go [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+import (
+	"gorm.io/gorm"
+)
 
 type User struct {
 	gorm.Model
 	SlackUserID string `gorm:"unique"`
 	Name        string
 	Avatar      string
+	Teams       []*Team `gorm:"many2many:team_users;"`
```
Comment (anirudhb) [inline pkg/db/users.go:12]: Don't think that this is really necessary. What would this be used for? (fwiw the insert above can just be replaced with a teams table insert if not exists)

**Review by anirudhb — CHANGES_REQUESTED** [review]
questions

File: pkg/db/users.go (hunk 2 above)
Comment (cjdenio) [reply pkg/db/users.go:12]: It's not necessary, but it's sort of a nicer-looking alternative to manually creating a team and assigning it to the user (https://gorm.io/docs/associations.html#Auto-Create-Update)

Comment (anirudhb) [reply pkg/db/users.go:12]: This is fine if you can rewrite GET /api/users/teams to use this as well, instead of the join.

**Review by anirudhb — APPROVED** [review]
lgtm

**Comment by cjdenio** [comment]
Awesome, just gonna reset the database before merging

**Pull request merged**